cadcetrans/cadcetrans/data_verify.py

```diff
@@ -0,0 +1,262 @@
+#  modify it under the terms of         modifier suivant les termes de
+#  the GNU Affero General Public        la “GNU Affero General Public
+#  License as published by the          License” telle que publiée
+#  Free Software Foundation,            par la Free Software Foundation
+#  either version 3 of the              : soit la version 3 de cette
+#  License, or (at your option)         licence, soit (à votre gré)
+#  any later version.                   toute version ultérieure.
+#
+#  OpenCADC is distributed in the       OpenCADC est distribué
+#  hope that it will be useful,         dans l’espoir qu’il vous
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+#  without even the implied             GARANTIE : sans même la garantie
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#
+# ***********************************************************************
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from .utils import etrans_config
+import logging
+import subprocess
+from hashlib import md5
+import re
+from PIL import Image
+import os
+import tarfile
+import tempfile
+from cadcetrans.utils import TransferException
+
+logger = logging.getLogger(__name__)
+
+fitsverify_output = re.compile(r' (\d+) warnings and (\d+) errors')
+
+
+def get_md5sum(filename):
+    """Return the MD5 sum of a file."""
+    sum = md5()
+    with open(filename, 'rb') as f:
+        while True:
+            data = f.read(1024)
+            if len(data) == 0:
+                break
+            sum.update(data)
+    return sum.hexdigest()
+
+
+def check_valid(filename, allow_warnings=True):
+    _, file_extension = os.path.splitext(filename)
+
+    method_name = etrans_config.get('verifiers', file_extension.strip('.'))
+    if not method_name:
+        logger.warning('No verifier found for file {}'.format(filename))
+        return
+    try:
+        method = globals()[method_name]
+    except KeyError:
+        logger.warning('Configured verifier for file extension {} not found. '
+                       'Check your config file.'.format(file_extension))
+        return
+    method(filename, allow_warnings)
+
+
+def check_valid_fits(filename, allow_warnings=True):
```

Comment:
Are we depending on clients having `fitsverify` available?  This looks like it will die if it's not installed.  I see the test checks for it first.

Reply:
Yes, it depends on the native fitsverify library.